### DISCUSSION TITLE
*done* error: Package subpath './public/extractFiles' is not defined by "exports" in C:\wiki\node_modules\extract-files\package.json

### DISCUSSION THREAD

**Discussion by SvenBastelt** [requarks/wiki#6656] [Help / Questions]
Hello, I use google translator because my English is not good enough to describe the problem.
Server 2016 Standard (all updates)
Wiki.js - Wiki.js 2.5.300
Node.js - V18.17.1 or v20.5.1
After setting up an SQL database without instructions, I start the whole thing with Powershell: node.server

PS C:\wiki> node server
Loading configuration from C:\wiki\config.yml... OK
2023-08-17T21:30:38.560Z [MASTER] info: =======================================
2023-08-17T21:30:38.561Z [MASTER] info: = Wiki.js 2.5.300 =====================
2023-08-17T21:30:38.562Z [MASTER] info: =======================================
2023-08-17T21:30:38.562Z [MASTER] info: Initializing...
2023-08-17T21:30:39.046Z [MASTER] info: Using database driver pg for postgres [ OK ]
2023-08-17T21:30:39.048Z [MASTER] info: Connecting to database...
2023-08-17T21:30:39.083Z [MASTER] info: Database Connection Successful [ OK ]
2023-08-17T21:30:39.593Z [MASTER] warn: Mail is not setup! Please set the configuration in the administration area!
2023-08-17T21:30:39.675Z [MASTER] info: Loading GraphQL Schema...
2023-08-17T21:30:40.124Z [MASTER] error: Package subpath './public/extractFiles' is not defined by "exports" in C:\wiki\node_modules\extract-files\package.json
PS C:\wiki>

As I said, I had installed the latest version, then read here very often, please read the requirements, there it says since version 2.5.300 Node.js can be used in 18 or later and also 20 or later. I also found the tip to insert 3 lines in the file, then I get a different error message.
I have no idea how to continue now.
I don't need a hint that the node version is not supported, because according to the wiki it is supported, but if I need a different version then please specify which version I have to use.
If that is the solution with the 3 lines, I ask for a screenshot of what it should look like in Notepad++.

Thanks in advance, SvenBastelt

PS: I've really done various searches here, but unfortunately never found a solution that I understood. sorry

**Comment by SvenBastelt** [comment]
here is the answer i search:

"exports": {
    ".": {
      "import": "./public/index.mjs",
      "require": "./public/index.js"
    },
    "./public/": "./public/",
    "./public/extractFiles": "./public/extractFiles.js",
    "./public/isExtractableFile": "./public/isExtractableFile.js",
    "./public/ReactNativeFile": "./public/ReactNativeFile.js",
    "./package": "./package.json",
    "./package.json": "./package.json"
	},

the question can be closed.

**Comment by ramanaidu214** [comment]
Hi , i am also facing same issue , even i added above exports section in package.json file , 
but still facing same issue , can you guide me how I can resolve this issue 
@SvenBastelt

**Reply by SvenBastelt** [reply]
go to this folder
...wiki...
\node_modules
\extract_Files
then open the file package.json with your Editor, i use notepad++
scroll to line 38, click in the front from this line and press enter
you have now a empty line 38

Copy this:

"./public/extractFiles": "./public/extractFiles.js",
"./public/isExtractableFile": "./public/isExtractableFile.js",
"./public/ReactNativeFile": "./public/ReactNativeFile.js",

an put in in your file on the line 38
after this you can for the optic the rowe 39 + 40 with space slide to the right

normaly it show so:

"./public/": "./public/",
"./public/extractFiles": "./public/extractFiles.js",
"./public/isExtractableFile": "./public/isExtractableFile.js",
"./public/ReactNativeFile": "./public/ReactNativeFile.js",
"./package": "./package.json",
"./package.json": "./package.json"
},

Save this file, restart in powershell and have fun.

I hope yor wiki is starting now.

**Reply by sheecet** [reply]
legend!

**Reply by VIT-JNE** [reply]
> go to this folder ...wiki... \node_modules \extract_Files then open the file package.json with your Editor, i use notepad++ scroll to line 38, click in the front from this line and press enter you have now a empty line 38
> 
> Copy this:
> 
> "./public/extractFiles": "./public/extractFiles.js", "./public/isExtractableFile": "./public/isExtractableFile.js", "./public/ReactNativeFile": "./public/ReactNativeFile.js",
> 
> an put in in your file on the line 38 after this you can for the optic the rowe 39 + 40 with space slide to the right
> 
> normaly it show so:
> 
> "./public/": "./public/", "./public/extractFiles": "./public/extractFiles.js", "./public/isExtractableFile": "./public/isExtractableFile.js", "./public/ReactNativeFile": "./public/ReactNativeFile.js", "./package": "./package.json", "./package.json": "./package.json" },
> 
> Save this file, restart in powershell and have fun.
> 
> I hope yor wiki is starting now.

Thank youuuuuu. That fixed the issue for me too.
When I did
journalctl -xe
after this, I only had problems left with SSL/HTTPS. But since this machine is gladly cut-off in an on-premise-environment, we can bear with that and just turned off SSL/HTTPS in the main config.yml.

This should be the main answer to a looot of points about "error: Package subpath './public/extractFiles' is not defined by "exports" in [...]\package.json", apart from Version-Rollbacks. For Windows AND for Debian/Ubuntu Linux.

Enjoy your week, everyone.

Best regards

**Reply by Najafov007** [reply]
> go to this folder ...wiki... \node_modules \extract_Files then open the file package.json with your Editor, i use notepad++ scroll to line 38, click in the front from this line and press enter you have now a empty line 38
> 
> Copy this:
> 
> "./public/extractFiles": "./public/extractFiles.js", "./public/isExtractableFile": "./public/isExtractableFile.js", "./public/ReactNativeFile": "./public/ReactNativeFile.js",
> 
> an put in in your file on the line 38 after this you can for the optic the rowe 39 + 40 with space slide to the right
> 
> normaly it show so:
> 
> "./public/": "./public/", "./public/extractFiles": "./public/extractFiles.js", "./public/isExtractableFile": "./public/isExtractableFile.js", "./public/ReactNativeFile": "./public/ReactNativeFile.js", "./package": "./package.json", "./package.json": "./package.json" },
> 
> Save this file, restart in powershell and have fun.
> 
> I hope yor wiki is starting now.

Hi, could you also please help me too, if it's okay(( Facing the EACESS issue - https://github.com/requarks/wiki/discussions/7443
